packages/flutter_tools/lib/src/resident_runner.dart

```diff
@@ -782,9 +782,9 @@ abstract class ResidentRunner {
     @required bool swap,
   }) {
     if (!fullRestart) {
-      return '$mainPath.incremental.dill';
+      return 'main.dart.incremental.dill';
```

Comment:
Does this cause any problems if the entrypoint is not called main.dart?

Reply:
Nope, the file name doesn't really matter - I tested with `test/hello_world_test.dart` and `main_2.dart`. It did historically when reload worked from source files.